File: Snooper/src/main/java/com/prateekj/snooper/networksnooper/views/HttpCallView.java
package com.prateekj.snooper.networksnooper.views;


import androidx.fragment.app.Fragment;

public interface HttpCallView {
  void copyToClipboard(String data);
  void shareData(String completeHttpCallData);
  void showMessageShareNotAvailable();
  Fragment getRequestBodyFragment();
  Fragment getResponseBodyFragment();
  Fragment getHeadersFragment();
  Fragment getExceptionFragment();
}

File: Snooper/src/main/java/com/prateekj/snooper/networksnooper/views/HttpCallView.kt
package com.prateekj.snooper.networksnooper.views


import androidx.fragment.app.Fragment

interface HttpCallView {
  fun getRequestBodyFragment(): Fragment
  fun getResponseBodyFragment(): Fragment
  fun getHeadersFragment(): Fragment
  fun getExceptionFragment(): Fragment
  fun copyToClipboard(data: String)
  fun shareData(completeHttpCallData: String)
  fun showMessageShareNotAvailable()
}
